crates/ibc/src/core/ics04_channel/handler/send_packet.rs

```diff
@@ -19,7 +19,10 @@ pub struct SendPacketResult {
     pub commitment: PacketCommitment,
 }
 
-pub fn send_packet(ctx: &dyn ChannelReader, packet: Packet) -> HandlerResult<PacketResult, Error> {
+pub fn send_packet(
+    ctx: &dyn SendPacketReader,
```

Comment:
How about we make this generic instead?

Reply:
https://github.com/cosmos/ibc-rs/pull/244/commits/b81611ae8f321de85a229c344b6fdd1ccbd5663b